app/src/main/java/fr/free/nrw/commons/depictions/Media/DepictedImagesPresenter.java

```diff
@@ -0,0 +1,167 @@
+     */
+    @SuppressLint("CheckResult")
+    @Override
+    public void initList(String entityId) {
+        view.setLoadingStatus(true);
+        view.progressBarVisible(true);
+        view.setIsLastPage(false);
+        compositeDisposable.add(depictsClient.fetchImagesForDepictedItem(entityId, 25, 0)
+                .subscribeOn(ioScheduler)
+                .observeOn(mainThreadScheduler)
+                .timeout(TIMEOUT_SECONDS, TimeUnit.SECONDS)
+                .subscribe(this::handleSuccess, this::handleError));
+    }
+
+    /**
+     * Fetches more images for the item and adds it to the grid view adapter
+     */
+    @SuppressLint("CheckResult")
+    @Override
+    public void fetchMoreImages() {
+        view.progressBarVisible(true);
+        compositeDisposable.add(depictsClient.fetchImagesForDepictedItem(entityId, 25, queryList.size())
+                .subscribeOn(ioScheduler)
+                .observeOn(mainThreadScheduler)
+                .timeout(TIMEOUT_SECONDS, TimeUnit.SECONDS)
+                .subscribe(this::handlePaginationSuccess, this::handleError));
+    }
+
+    /**
+     * Handles the success scenario
+     * it initializes the recycler view by adding items to the adapter
+     */
+    private void handlePaginationSuccess(List<Media> media) {
+        queryList.addAll(media);
+        view.progressBarVisible(false);
+        view.addItemsToAdapter(media);
+    }
+
+    /**
+     * Logs and handles API error scenario
+     *
+     * @param throwable
+     */
+    public void handleError(Throwable throwable) {
+        Timber.e(throwable, "Error occurred while loading images inside items");
+        try {
+            view.initErrorView();
+            view.showSnackBar();
+        } catch (Exception e) {
+            e.printStackTrace();
+        }
+
+    }
+
+    /**
+     * Handles the success scenario
+     * On first load, it initializes the grid view. On subsequent loads, it adds items to the adapter
+     * @param collection List of new Media to be displayed
+     */
+    public void handleSuccess(List<Media> collection) {
+        if (collection == null || collection.isEmpty()) {
+            if (queryList.isEmpty()) {
+                view.initErrorView();
+            } else {
+                view.setIsLastPage(true);
+            }
+        } else {
+            this.queryList.addAll(collection);
+            view.handleSuccess(collection);
+        }
+    }
+
+    /**
+     * fetch captions for the image using filename and replace title of on the image thumbnail(if captions are available)
+     * else show filename
+     */
+    @Override
+    public void replaceTitlesWithCaptions(String wikibaseIdentifier, int position) {
+        compositeDisposable.add(mediaClient.getCaptionByWikibaseIdentifier(wikibaseIdentifier)
+                .subscribeOn(ioScheduler)
+                .observeOn(mainThreadScheduler)
+                .timeout(TIMEOUT_SECONDS, TimeUnit.SECONDS)
```

Comment:
Should we consider replacing timeouts from over here, OKHTTP client does have timeout annyways

Reply:
happy to